src/ipfs/daemon.rs

```diff
@@ -0,0 +1,221 @@
+                bail!(
+                    "{}",
+                    "Failed to start ipfs because the timeout reached!!".red()
+                )
+            }
+        }
+        anyhow::Ok(())
+    }
+}
+
+impl TryFrom<IpfsConnInfo> for IpfsDaemon{
+    type Error = anyhow::Error;
+    /// This is one of the two main ways to make a new instance of the IPFS daemon struct. Use this
+    /// method when you want to make a new instance and you need to set the port and address of
+    /// the daemon you are attempting to refrence
+    fn try_from(conn_info: IpfsConnInfo) -> Result<Self> {
+        let client = IpfsClient::from_host_and_port(
+            "http".parse()?, 
+            &conn_info.address, 
+            conn_info.port
+        )?;
+
+        let runtime = tokio::runtime::Runtime::new()?;
+        Ok(Self{
+            client,
+            tokio: runtime,
+            conn_info
+        })   
+    }
+}
+
+impl Default for IpfsDaemon {
+    /// This is one of the two main ways to make a new instance of the IPFS daemon struct. Use this
+    /// method when you want to make a new instance and you okay with using the default port and address
+    /// set by the config.
+    /// 
+    /// Note: This method has the possiblity of throwing an error if the configuration of the port and
+    /// address is not setup right.
+    fn default() -> Self {
+        Self::try_from(IpfsConnInfo{
+            address:IPFS_ADDR.to_string(),
+            port:IPFS_API_PORT
+        }).unwrap()
+    }
+}
+impl Clone for IpfsDaemon {
+    fn clone(&self) -> Self {
+        return Self { client: self.client.clone(), tokio: Runtime::new().unwrap(), conn_info: self.conn_info.clone()}
+    }
+}
+
+#[async_trait]
+impl Ipfs for IpfsDaemon {
+    async fn add(&self, path: &Path) -> Result<HashMap<String, String>> {
+        let response_list = self.tokio.block_on(async {
+            self.client.add_path(path).await
+        })?;
+        return Ok(response_list.into_iter().map(|res| {
+            (res.name, res.hash)
+        }).collect());
+    }
+    async fn connect_to(&self, peer_id: &str) -> Result<()> {
+        let messages =  self.tokio.block_on(async {
+            self.client.swarm_connect(peer_id).await
+        })?.strings;
+        for msg in messages {
+            println!("{}", msg.blue());
+            if !msg.contains("success"){
+                bail!(msg);
+            }
+        }
+        return Ok(());
+    }
+    async fn get_connected(&self) -> Result<Vec<String>> {
+        let peers =  self.tokio.block_on(async {
+            self.client.swarm_peers().await
+        })?;
+        Ok(peers.peers.into_iter().map(|peer| peer.addr).collect())
+    }
+    async fn get_config(&self, prop:&str) -> Result<Value>{
+        let config = self.tokio.block_on(async {
+            self.client.config_get_json(prop).await
+        })?;
+        return Ok(config.value);
+    }
+    async fn set_config(&self, prop:&str, val:&Value) -> Result<()> {
```

Comment:
We may want to check for invalid JSON values. What happens if `null` or a number are passed to this function?